vaadin-spring/src/main/java/com/vaadin/flow/spring/SpringBootAutoConfiguration.java

```diff
@@ -93,20 +94,22 @@ public ServletContextInitializer contextInitializer() {
     }
 
     /**
-     * Creates a {@link DispatcherServletRegistrationBean} instance for a dispatcher
-     * servlet in case Vaadin servlet is mapped to the root.
+     * Creates a {@link DispatcherServletRegistrationBean} instance for a
+     * dispatcher servlet in case Vaadin servlet is mapped to the root.
      * <p>
      * This is needed for correct servlet path (and path info) values available
      * in Vaadin servlet because it works via forwarding controller which is not
      * properly mapped without this registration.
      *
-     * @return a custom DispatcherServletRegistrationBean instance for dispatcher servlet
+     * @return a custom DispatcherServletRegistrationBean instance for
+     *         dispatcher servlet
      */
     @Bean
     @Conditional(RootMappedCondition.class)
-    public DispatcherServletRegistrationBean dispatcherServletRegistration() {
+    @ConditionalOnMissingClass("org.springframework.boot.autoconfigure.web.servlet.DispatcherServletRegistrationBean")
+    public ServletRegistrationBean<DispatcherServlet> dispatcherServletRegistration() {
```

Comment:
As there is a extra condition, it would be worth adding a note as why it is necessary?